pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/xpath/impl/AttributeAxisIterator.java

```diff
@@ -119,7 +122,23 @@ && isConsideredReturnType(method)
 
     private boolean isConsideredReturnType(Method method) {
         Class<?> klass = method.getReturnType();
-        return CONSIDERED_RETURN_TYPES.contains(klass) || klass.isEnum();
+        if (CONSIDERED_RETURN_TYPES.contains(klass) || klass.isEnum()) {
+            return true;
+        }
+
+        if (Collection.class.isAssignableFrom(klass)) {
+            Type t = method.getGenericReturnType();
+            if (t instanceof ParameterizedType) {
+                try {
+                    Class<?> elementKlass = Class.forName(((ParameterizedType) t).getActualTypeArguments()[0].getTypeName());
+                    return CONSIDERED_RETURN_TYPES.contains(elementKlass) || elementKlass.isEnum();
+                } catch (ClassNotFoundException ignored) {
+                    // should never happen
```

Comment:
Even if this should never happen, can we still log it as error, so that - if it happens anyway (e.g. due to JVM bug or other maybe temporary IO issue) we are aware of it?

Reply:
Done!